crates/fullnode/src/da_block_handler.rs

```diff
@@ -847,12 +845,10 @@ where
         &self,
         current_l1_block_height: u64,
     ) -> Result<(), ProcessingError> {
-        let pending_proofs = self.ledger_db.get_pending_proofs()?;
-        if pending_proofs.is_empty() {
-            return Ok(());
-        }
+        let mut pending_proofs = self.ledger_db.get_pending_proofs()?;
 
-        for ((min_index, max_index), proof, found_in_l1_height) in pending_proofs {
+        while let Some(Ok(item)) = pending_proofs.next() {
```

Comment:
There is `impl<S> Iterator for SchemaIterator<'_, S>`, so I don't see why we can't use for instead of while let some.